tests/pytests/unit/states/test_gem.py

```diff
@@ -0,0 +1,137 @@
+"""
+    Tests of salt.states.gem
+"""
+
+
+import pytest
+
+import salt.states.gem as gem
+from tests.support.mock import MagicMock, patch
+
+
+@pytest.fixture
+def configure_loader_modules():
+    return {gem: {"__opts__": {"test": False}}}
+
+
+def test_installed():
+    gems = {"foo": ["1.0"], "bar": ["2.0"]}
+    gem_list = MagicMock(return_value=gems)
+    gem_install_succeeds = MagicMock(return_value=True)
+    gem_install_fails = MagicMock(return_value=False)
+
+    with patch.dict(gem.__salt__, {"gem.list": gem_list}):
+        with patch.dict(gem.__salt__, {"gem.install": gem_install_succeeds}):
+            ret = gem.installed("foo")
+            assert ret["result"] is True
+            ret = gem.installed("quux")
+            assert ret["result"] is True
+            gem_install_succeeds.assert_called_once_with(
+                "quux",
+                pre_releases=False,
+                ruby=None,
+                runas=None,
+                version=None,
+                proxy=None,
+                rdoc=False,
+                source=None,
+                ri=False,
+                gem_bin=None,
+            )
+
+        with patch.dict(gem.__salt__, {"gem.install": gem_install_fails}):
+            ret = gem.installed("quux")
+            assert ret["result"] is False
+            gem_install_fails.assert_called_once_with(
+                "quux",
+                pre_releases=False,
+                ruby=None,
+                runas=None,
+                version=None,
+                proxy=None,
+                rdoc=False,
+                source=None,
+                ri=False,
+                gem_bin=None,
+            )
+
+
+def test_installed_version():
+    gems = {"foo": ["1.0"], "bar": ["2.0"]}
+    gem_list = MagicMock(return_value=gems)
+    gem_install_succeeds = MagicMock(return_value=True)
+
+    with patch.dict(gem.__salt__, {"gem.list": gem_list}):
+        with patch.dict(gem.__salt__, {"gem.install": gem_install_succeeds}):
+            ret = gem.installed("foo", version=">= 1.0")
+            assert ret["result"] is True
+            assert "Installed Gem meets version requirements." == ret["comment"]
```

Comment:
```suggestion
            assert ret["comment"] == "Installed Gem meets version requirements."
```
Assert what we get against what we expect